### ISSUE TITLE
Missing ES2PhongShader

### ISSUE THREAD

**Issue by mipastgt** [gluonhq/client-maven-plugin#112]
In a test program where I want to display a TriangleMesh I always get the following exception.
```
java.lang.NoClassDefFoundError: Could not initialize class com.sun.prism.es2.ES2PhongShader
	at com.oracle.svm.core.hub.ClassInitializationInfo.initialize(ClassInitializationInfo.java:214)
	at java.lang.Class.ensureInitialized(DynamicHub.java:475)
	at com.sun.prism.es2.ES2Context.getPhongShader(ES2Context.java:142)
	at com.sun.prism.es2.ES2Context.renderMeshView(ES2Context.java:471)
	at com.sun.prism.es2.ES2MeshView.render(ES2MeshView.java:119)
	at com.sun.javafx.sg.prism.NGShape3D.renderMeshView(NGShape3D.java:179)
	at com.sun.javafx.sg.prism.NGShape3D.renderContent(NGShape3D.java:201)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGSubScene.renderContent(NGSubScene.java:218)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGRegion.renderContent(NGRegion.java:579)
	at com.sun.javafx.sg.prism.NGNode.renderForClip(NGNode.java:2313)
	at com.sun.javafx.sg.prism.NGNode.renderRectClip(NGNode.java:2207)
	at com.sun.javafx.sg.prism.NGNode.renderClip(NGNode.java:2233)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2066)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGRegion.renderContent(NGRegion.java:579)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.sg.prism.NGGroup.renderContent(NGGroup.java:270)
	at com.sun.javafx.sg.prism.NGRegion.renderContent(NGRegion.java:579)
	at com.sun.javafx.sg.prism.NGNode.doRender(NGNode.java:2072)
	at com.sun.javafx.sg.prism.NGNode.render(NGNode.java:1964)
	at com.sun.javafx.tk.quantum.ViewPainter.doPaint(ViewPainter.java:479)
	at com.sun.javafx.tk.quantum.ViewPainter.paintImpl(ViewPainter.java:321)
	at com.sun.javafx.tk.quantum.PresentingPainter.run(PresentingPainter.java:91)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at com.sun.javafx.tk.RenderJob.run(RenderJob.java:58)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:125)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:497)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)

```
I already tried to add the class to the reflectionList and the jniList but this did not make any difference.

Is 3D rendering not supported yet? 

Environment:
macOS 10.14.6
GraalVM CE 11 19.3.1
OpenJFX 14-ea+7
client-maven-plugin 0.1.12

**Comment by jperedadnr** [comment]
3D is supported.

Precisely that class was added to Substrate with this commit: https://github.com/gluonhq/substrate/issues/262, right before releasing the client plugin 0.0.12

Anyway, as you can see, the class was added to the reflection list.

**Comment by mipastgt** [comment]
This is strange because then it should work without manual intervention by me but it doesn't.
The whole issue is strange anyway because in order to get my program that far I have to add several classes to the reflectionList in the POM manually although some of them are then duplicates in the generated reflection list (see attachment) but without adding them I get errors like the above. It seems to me that the default entries are not complete enough, or something like that. The only class where this kind of overwriting does not work is the above mentioned ES2PhongShader. Maybe because it is a `com.sun.*` class?
```
<reflectionList>
    <list>com.sun.prism.es2.ES2PhongShader</list>
    <list>javafx.scene.LightBase</list>
    <list>javafx.scene.AmbientLight</list>
    <list>javafx.scene.Group</list>
    <list>javafx.scene.shape.SVGPath</list>
    <list>javafx.scene.shape.Polyline</list>
    <list>javafx.scene.shape.Polygon</list>
    <list>javafx.scene.paint.Color</list>
    <list>javafx.fxml.FXMLLoader</list>
    <list>de.mpmediasoft.mpmappanefx.core.trafo.Viewport</list>
    <list>de.mpmediasoft.mpmappanefx.gui.map.MapContent</list>
    <list>de.mpmediasoft.mpmappanefx.gui.layers.tiles.tilesources.OpenFlightMapsOsmTileSource</list>
</reflectionList>
```
The default list seems to be incomplete anyway. For example you have Polygon in there but not Polyline or is that list somehow filtered before it is emitted.

Maybe I should add that, if I eliminate the 3D portion of my program, the rest of the map rendering now works perfectly with good performance and blazingly fast start-up time.

[reflectionconfig-x86_64-darwin.json.zip](https://github.com/gluonhq/client-maven-plugin/files/4090483/reflectionconfig-x86_64-darwin.json.zip)

**Comment by mipastgt** [comment]
Ha, I have found and solved the problem. The bug is in your default resource config. You are missing an entry like `{"pattern": ".*\\.vert$"}` .
The ES2PhongShader loads the following resource in addition to all the `*.frag` files, for which you have an entry.
```
vertexShaderSource = ES2Shader.readStreamIntoString(ES2ResourceFactory.class.getResourceAsStream("glsl/main.vert"));

```
So, there are actually two problems here: 1. The error message is completely misleading because it says `NoClassDefFoundError` and 2. you have to add this entry for `*.vert` files. It would also be nice to add an entry for `*.properties` files because they are so common.
As far as I can see the ES2PhongShader has never worked unless someone has manually specified the missing resource pattern.

My map rendering works perfectly now. Me happy !!!

**Comment by jperedadnr** [comment]
Right, you need that extension too... it was already in the pipeline: https://github.com/gluonhq/substrate/pull/283

The problem with `*.properties` is that in many cases these can be bundle resource files, and you have to add them to the bundlesList.

**Comment by jperedadnr** [comment]
Fixed with plugin 0.1.15
